sqlmesh/core/config/format.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+import typing as t
+
+from pydantic import Field
+
+from sqlmesh.core.config.base import BaseConfig
+
+
+class FormatConfig(BaseConfig):
+    """The format configuration for SQL code.
+
+    Args:
+        append_newline: Whether to append a newline to the end of the file or not.
+        normalize: Whether to normalize the SQL code or not.
+        pad: The number of spaces to use for padding.
+        indent: The number of spaces to use for indentation.
+        normalize_functions: The functions to normalize.
+        leading_comma: Whether to use leading commas or not.
+        max_text_width: The maximum text width.
+    """
+
+    append_newline: bool = Field(default=False, exclude=True)
```

Comment:
Why do we need `exclude=False` here? I think this is what trips up Pydantic V1

Reply:
Because everything is a kwarg to the `Generator` class except this option which is handled in sqlmesh side @izeigerman

Reply:
So it simplifies dumping to dict and passing through to `format_model_expressions` or w/e

Reply:
See here:
https://github.com/z3z1ma/sqlmesh/blob/4f8d2fb29afd511ca5db4a847f964555dfd539d8/sqlmesh/core/context.py#L766-L768
https://github.com/z3z1ma/sqlmesh/blob/4f8d2fb29afd511ca5db4a847f964555dfd539d8/web/server/api/endpoints/files.py#L81

where we would otherwise need to remember to pass `exclude={"append_newline"}` to the dict call to be compatible with `Generator` @izeigerman 
The pydantic v1 error may make it inevitable, so its nbd

Reply:
Okay i just added a dedicated prop to solve for this instead of implicit pydantic behavior
tbd if you want to change the name